test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -298,10 +298,38 @@ BOOST_AUTO_TEST_CASE(C99_scoping_activation)
 				}
 				return x;
 			}
+			function g() pure public returns (uint x) {
+				x = 7;
+				{
+					x = 3;
+					uint x;
+					return x; // This returns the new variable, i.e. 0
+				}
+			}
+			function h() pure public returns (uint x, uint a, uint b) {
+				x = 7;
+				{
+					x = 3;
+					a = x; // This should read from the outer
+					uint x = 4;
+					b = x;
+				}
+			}
+			function i() pure public returns (uint x, uint a) {
+				x = 7;
+				{
+					x = 3;
+					uint x = x; // This should read from the outer and assign to the inner
```

Comment:
That is super convoluted :)

Reply:
I take that as a compliment :)
It will generate a shadowing warning, so I think we do not need to make it more strict.